File: backend/app/schemas/project.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, Any


class ProjectCreate(BaseModel):
    key: Optional[str] = None
    name: str
    description: Optional[str] = None
    teamId: Optional[str] = None
    settings: Optional[dict[str, Any]] = None


class ProjectUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    status: Optional[str] = None
    leadId: Optional[str] = None
    settings: Optional[dict[str, Any]] = None


class ProjectResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    key: str
    name: str
    description: Optional[str] = None
    status: str
    leadId: Optional[str] = None
    teamId: Optional[str] = None
    settings: str
    createdById: str
    createdAt: str
    updatedAt: str

File: frontend/src/types/project.ts
export interface Project {
  id: string;
  key: string;
  name: string;
  description: string | null;
  status: string;
  visibility: string;
  leadId: string | null;
  teamId: string | null;
  settings: Record<string, unknown>;
  createdById: string;
  createdAt: string;
  updatedAt: string;
  deletedAt: string | null;
}

export interface ProjectStats {
  total: number;
  open: number;
  closed: number;
  backlog: number;
}

export interface ProjectsResponse {
  projects: Project[];
  pagination: {
    page: number;
    limit: number;
    total: number;
    totalPages: number;
  };
}

export interface Workflow {
  id: string;
  projectId: string;
  name: string;
  statuses: string[];
  transitions: Record<string, string[]>;
  isDefault: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface Label {
  id: string;
  projectId: string;
  name: string;
  color: string;
  createdAt: string;
}
